### ISSUE TITLE
implicit Read[Option[A]] - compilation times

### ISSUE THREAD

**Issue by decoursin** [typelevel/doobie#1054]
Using `scalac-profiling`, I was able to generate a flamegraph to profile the compilation of my scala project. With great success, I was able to dramatically reduce compilation times by creating implicit `Read`s for my scala composite case classes.

However, it seems I'm not able to do this for `Option` types.

If I have some composite type `A`, the `Read[Option[A]]` is automatically generated by Doobie. Which is cool, except that it seems there's no way for me to write `Read[Option[A]]` myself to avoid the extra compilation process.

Even when I've created the `Read[A]` myself, it seemingly doesn't effect the compilation times of generating `Read[Option[A]]`.

Is it possible to manually write the `Read[Option[A]]`, if so, how?

To summarize, my flamegraph says that:
```
doobie.util.Read[Option[com.panono.db.DBAlbum]] (expanded macros 0)  (2,546,205 Î¼s, 5.65%)
```
even when there exists a `Read[DBAlbum]`.

**Comment by jatcwang** [comment]
@decoursin I think putting a `implicit val readOptDbAlbum: Read[Option[DbAlbum]] = ....` in `DBAlbum`'s companion object should work?

**Comment by decoursin** [comment]
@jatcwang Thanks
